imap/search_xapian.c

```diff
@@ -2658,6 +2660,18 @@ static int begin_mailbox_update(search_text_receiver_t *rx,
         goto out;
     }
 
+    /* make sure the conversations are open before we start indexing
+     * to avoid deadlocking against the search state */
+    if (tr->mode == XAPIAN_DBW_CONVINDEXED) {
+        struct conversations_state *cstate = mailbox_get_cstate(tr->super.mailbox);
```

Comment:
This should get the cstate from `mailbox` instead of `tr->super.mailbox`. The latter only is set at the end of this function.

Reply:
Yeah!  Duh.  Fixed.